src/steps/campaign-member-field-equals.ts

```diff
@@ -0,0 +1,70 @@
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class CampaignMemberFieldEquals extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check a Campaign Member Field Value';
+  /* tslint:disable-next-line:max-line-length */
+  protected stepExpression: string = 'the salesforce lead (?<email>.+) should be a member of campaign (?<campaignId>.+) with (?<field>.+) set to (?<expectedValue>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'email',
+    type: FieldDefinition.Type.EMAIL,
+    description: "CampaignMember's email address",
+  }, {
+    field: 'campaignId',
+    type: FieldDefinition.Type.STRING,
+    description: 'CampaignId',
+  }, {
+    field: 'field',
+    type: FieldDefinition.Type.STRING,
+    description: 'Field name to check',
+  }, {
+    field: 'expectedValue',
+    type: FieldDefinition.Type.ANYSCALAR,
+    description: 'Expected field value',
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const email: string = stepData.email;
+    const campaignId: string = stepData.campaignId;
+    const field: string = stepData.field;
+    const expectedValue: string = stepData.expectedValue;
+    let campaignMember: Record<string, any>;
+
+    try {
+      campaignMember = await this.client.findCampaignMemberByEmail(email, [field, 'CampaignId']);
+    } catch (e) {
+      return this.error('There was a problem checking the CampaignMember: %s', [e.toString()]);
+    }
+
+    if (!campaignMember) {
+      // If no results were found, return an error.
+      return this.error('No CampaignMember found with email %s', [email]);
+    } else if (campaignMember['CampaignId'] !== campaignId) {
```

Comment:
Done. Removed that error; i just included that campaignId error to the campaignMember error itself in the message since if they cant find a member with both data, it wont return any.